ci/deps.sh

```diff
@@ -20,9 +20,14 @@
 #Do not delete
 set -e
 
-#wget Retries
+#wget options
 WGET_N_RETRIES=30
 WGET_WAIT_RETRIES_S=10
+WGET_FLAGS="-t $WGET_N_RETRIES --waitretry=$WGET_WAIT_RETRIES_S"
+DEPS_DONT_CHECK_CERTIFICATE="${DEPS_DONT_CHECK_CERTIFICATE:-}"
```

Comment:
Hopefully the last comment. Seems this doesn't work without `bash`/`dash` (albeit it worked in `ksh`

With `tcsh` it doesn't, so I suspect it's not "safe" (`sh` portability):

```
root@9f5bc224f6b2:/# cat test.sh 
#!/bin/sh
TEST=" ffff "
TEST2=${TEST:-}
echo "$TEST2"
root@9f5bc224f6b2:/# rm /bin/sh 
root@9f5bc224f6b2:/# ln -s /bin/tcsh /bin/sh
root@9f5bc224f6b2:/# sh test.sh 
TEST= ffff : Command not found.
Bad : modifier in $ '-'.
```

Reply:
Good catch. Yes, I'm maybe too used to bash functionality, and I don't think defining vars with defaults like this is supported in many other shells. However, that line is redundant (as taking the value of a var, or an empty value as a default, is almost the same than just picking the env var directly), but I tend to incorporate this even in this scenario, to make that env var value recollection/usage explicit and clear at the beginning of the script (rather than finding out the usage of custom env vars in the middle of scripts) plus, if my memory is not wrong, that also lets you use the "bash nounset mode" (`-u`) in case the env var is not defined.

So I'm going to just drop that line: I guess the script is not long, and the value of `DEPS_DONT_CHECK_CERTIFICATE` is picked quite early in the script, so an user of this will quickly identify that functionality (if we ever want to). But, anyways, my very personal opinion about these topics with this script (and it's an opinion from someone with very little context, so it has relative value): I think the cost of trying to maintain an script that installs deps in very diverse environments is quite high... I would consider (if possible) that, for users needing to install pmacct in a system without bash, let the person interested on that to care about deps installation, or to install bash in their system, rather than trying to consider this an "universal script" and assume the extra maintenance cost. And if you really want to keep property, maybe CI can do these compat checks for you (for instance, in some software/scripts I maintain which are expected to be compat with osx, I tend to run them on top of some deprecated base images that come with ancient versions of bash (ubuntu precise?), mimicking osx versioning... maybe you can find or create some base images that can represent your target environments)

Reply:
> would consider (if possible) that, for users needing to install pmacct in a system without bash, let the person interested on that to care about deps installation, or to install bash in their system, rather than trying to consider this an "universal script" and assume the extra maintenance cost. 

This script is only used by the CI a) build checks b) docker

Reply:
Got it. So my very personal recommendation: current CI coverage then wouldn't break by using bash, and if you ever have more "build shapes" / release this software as other artifact types with more builders/base images, I guess bash could be also installed in these other Unix-like systems, so I would consider just embracing bash(4?). But, in any case, that was an almost off-topic conversation: we are back to sh-compat, so marking this as resolved.